src/cyclonedx/agent/cyclonedx.php

```diff
@@ -0,0 +1,535 @@
+
+    $packageNodes = $this->renderPackage($uploadId);
+
+    $this->computeUri($uploadId);
+
+    $contents = json_encode($packageNodes, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+
+    $this->writeReport($contents, $uploadId);
+    return true;
+  }
+
+  /**
+   * @brief Get the URI for the given package
+   * @param string $fileBase Name of the upload
+   * @return string URI for the upload
+   */
+  protected function getUri($fileBase)
+  {
+    $fileName = $fileBase. strtoupper($this->outputFormat)."_".$this->packageName.'_'.date("Y-m-d_H:i:s");
+    $fileName = $fileName .".json" ;
+    return $fileName;
+  }
+
+  /**
+   * @brief For a given upload, compute the URI
+   * @param int $uploadId
+   */
+  protected function computeUri($uploadId)
+  {
+    global $SysConf;
+    $upload = $this->uploadDao->getUpload($uploadId);
+    $this->packageName = $upload->getFilename();
+
+    $fileBase = $SysConf['FOSSOLOGY']['path']."/report/";
+
+    $this->uri = $this->getUri($fileBase);
+  }
+
+  /**
+   * @brief Given an upload id, render the report string
+   * @param int $uploadId
+   * @return string Rendered report string
+   */
+  protected function renderPackage($uploadId)
+  {
+    global $SysConf;
+    $uploadTreeTableName = $this->uploadDao->getUploadtreeTableName($uploadId);
+    $itemTreeBounds = $this->uploadDao->getParentItemBounds($uploadId,$uploadTreeTableName);
+    $this->heartbeat(0);
+
+    $filesWithLicenses = $this->getFilesWithLicensesFromClearings($itemTreeBounds);
+    $this->heartbeat(0);
+
+    $this->addClearingStatus($filesWithLicenses,$itemTreeBounds);
+    $this->heartbeat(0);
+
+    $this->addScannerResults($filesWithLicenses, $itemTreeBounds);
+    $this->heartbeat(0);
+
+    $this->addCopyrightResults($filesWithLicenses, $uploadId);
+    $this->heartbeat(0);
+
+    $upload = $this->uploadDao->getUpload($uploadId);
+    $components = $this->generateFileComponents($filesWithLicenses, $upload->getTreeTableName(), $uploadId, $itemTreeBounds);
+
+    $mainLicenseIds = $this->clearingDao->getMainLicenseIds($uploadId, $this->groupId);
+    $mainLicenses = array();
+    $licenseTexts = $this->getLicenseTexts();
+    foreach ($mainLicenseIds as $licId) {
+      $reportedLicenseId = $this->licenseMap->getProjectedId($licId);
+      $this->includedLicenseIds[$reportedLicenseId] = true;
+      $spdxId = $this->licenseMap->getProjectedSpdxId($reportedLicenseId);
+      $licensedata['id'] = $spdxId;
+      $licensedata['url'] = $licenseTexts[$spdxId]['url'];
+      $mainLicenses[] = $this->reportGenerator->createLicense($licensedata);
+    }
+
+    $hashes = $this->uploadDao->getUploadHashes($uploadId);
+    $serializedhash = array();
+    $serializedhash[] = $this->reportGenerator->createHash('SHA-1', $hashes['sha1']);
+    $serializedhash[] = $this->reportGenerator->createHash('MD5', $hashes['md5']);
+    $serializedhash[] = $this->reportGenerator->createHash('SHA-256', $hashes['sha256']);
```

Comment:
Please check if `sha256` is not empty before processing it as old uploads might not contain that value.